### PR TITLE
[utils.sh] Split `addOrEditKeyValuePair` to do `addKey` in a separate function

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Having a function with a variable number of arguments in this case was too complicated for something that is only really used once

### CODE REVIEW

File: pihole [hunk 1]
```diff
@@ -260,7 +260,7 @@ Options:
     exit 0
   elif [[ "${1}" == "off" ]]; then
     # Disable logging
-    addOrEditKeyValPair /etc/dnsmasq.d/01-pihole.conf "log-queries"
+    addKey /etc/dnsmasq.d/01-pihole.conf "log-queries"
```
Comment (yubiuser) [inline pihole:263]:
```suggestion
    remove /etc/dnsmasq.d/01-pihole.conf "log-queries"
```

L273 change `removeKey` to `addKey`

**Review by yubiuser — CHANGES_REQUESTED** [review]

**Review by yubiuser — CHANGES_REQUESTED** [review]
Something is wrong with the base branch of this branch.

```
rockpi@rockpi-4b:~$ pihole -a -p
/opt/pihole/webpage.sh: line 31: utilsfile: readonly variable
```

This is not in development
https://github.com/pi-hole/pi-hole/blob/9356d7bbb122f3c69fc97a2b3aff828944ef84d6/advanced/Scripts/webpage.sh#L31

But in this branch
https://github.com/pi-hole/pi-hole/blob/87757206688d4a2d9caa01e1e68accd0b51d2d4d/advanced/Scripts/webpage.sh#L31

However, it's not shown here in the Github diff.....

**Comment by PromoFaux** [comment]
Should I start again?

**Comment by PromoFaux** [comment]
https://github.com/pi-hole/pi-hole/pull/4698

**Pull request closed without merging**